src/main/java/rx/schedulers/JavaFxScheduler.java

```diff
@@ -46,86 +48,167 @@ public static JavaFxScheduler getInstance() {
+                            if (tracking.isUnsubscribed() || isUnsubscribed()) {
+                                return;
+                            }
+                            action.call();
+                        } finally {
+                            subs.unsubscribe();
+                        }
+                    } else {
+                        timeline.stop();
+                        timeline = null;
+                        nonDelayed = true;
+                        trampoline(this);
+                    }
+                }
 
-            // wrap for returning so it also removes it from the 'innerSubscription'
-            return Subscriptions.create(new Action0() {
+                @Override
+                public boolean isUnsubscribed() {
+                    return subs.isUnsubscribed();
+                }
 
                 @Override
-                public void call() {
-                    timeline.stop();
-                    s.unsubscribe();
-                    innerSubscription.remove(s);
+                public void unsubscribe() {
+                    subs.unsubscribe();
+                }
+                public void set(Subscription s) {
+                    subs.set(s);
                 }
+            }
 
-            });
+            final DualAction executeOnce = new DualAction();
+            tracking.add(executeOnce);
+
+            //final Timeline timer = new Timeline((int) delay, executeOnce);
+            final Timeline timer = new Timeline(new KeyFrame(Duration.millis(delay)));
+
+            executeOnce.setTimer(timer);
+            timer.play();
+
+            executeOnce.set(Subscriptions.create(() -> {
+                timer.stop();
+                tracking.remove(executeOnce);
+            }));
+
+            return executeOnce;
         }
 
         @Override
         public Subscription schedule(final Action0 action) {
             final BooleanSubscription s = BooleanSubscription.create();
-            Platform.runLater(new Runnable() {
-                @Override
-                public void run() {
-                    if (innerSubscription.isUnsubscribed() || s.isUnsubscribed()) {
-                        return;
-                    }
-                    action.call();
-                    innerSubscription.remove(s);
+            Runnable runnable = () -> {
+                if (tracking.isUnsubscribed() || s.isUnsubscribed()) {
+                    return;
                 }
-            });
+                action.call();
+                tracking.remove(s);
+            };
 
-            innerSubscription.add(s);
-            // wrap for returning so it also removes it from the 'innerSubscription'
-            return Subscriptions.create(new Action0() {
-
-                @Override
-                public void call() {
-                    s.unsubscribe();
-                    innerSubscription.remove(s);
+            if (Platform.isFxApplicationThread()) {
```

Comment:
What if `schedule()` is called from not the platform thread? Looks like you need:

```java
if (Platform.isFxApplicationThread() && trampoline(runnable)) {
    return Subscriptions.unsubscribed();
} else {
    queue.offer(runnable);
    Platform.runLater(this);
}
```